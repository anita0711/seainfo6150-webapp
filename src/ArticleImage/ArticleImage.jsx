import React from "react";
import PropTypes from "prop-types";
import styles from "./ArticleImage.module.css";

const ArticleImage = function (props) {
  return (
    <div>
      <img src={props.url} alt="" className={styles.image}></img>
      <h2 className={styles.title}>{props.title}</h2>
    </div>
  );
};

ArticleImage.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default ArticleImage;
