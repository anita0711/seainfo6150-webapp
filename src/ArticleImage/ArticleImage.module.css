.image {
  width: 50%;
  height: auto;
  border-radius: 50%;
  border: 10px solid #c00;
}

.title {
  color: #c00;
  margin: 0 0 20px;
  text-decoration: underline;
}
