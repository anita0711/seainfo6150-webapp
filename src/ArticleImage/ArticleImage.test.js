import React from "react";
import ArticleImage from "./ArticleImage";

describe("ArticleTextToggleButton tests", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ArticleImage
        url="https://cdn.cnn.com/cnnnext/dam/assets/181116131343-garima-arora-michelin-super-169.jpg"
        title="India's first woman with a Michelin star"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
