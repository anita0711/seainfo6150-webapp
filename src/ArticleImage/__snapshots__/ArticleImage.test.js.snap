// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      class="image"
      src="https://cdn.cnn.com/cnnnext/dam/assets/181116131343-garima-arora-michelin-super-169.jpg"
    />
    <h2
      class="title"
    >
      India's first woman with a Michelin star
    </h2>
  </div>
</div>
`;
